### ISSUE TITLE
Munin Node should provide more directories to plugins via environments

### ISSUE THREAD

**Issue by steveschnepp** [munin-monitoring/munin#949]
There is already several env vars that exists: 
* `MUNIN_STATEFILE`: per-plugin state file that remains across executions
* `MUNIN_PLUGSTATE`: state directory that is shared among all plugins, to be able to share

Yet, it would be nice to have also:
* `MUNIN_STATEDIR` that is a whole directory
* `MUNIN_TMPDIR` that is a temporary directory that is scoped to the plugin execution

The `MUNIN_TMPDIR` will ensure proper cleanup of the leftovers, and therefore avoid the need for http://guide.munin-monitoring.org/en/latest/develop/plugins/advanced-topics.html#handling-temporary-files hygiene.

**Comment by sumpfralle** [comment]
regarding `MUNIN_TMPDIR`: I would be a bit scared by the security implication of plugins not using "mktemp", but something self-made (e.g. "echo >foo").
Thus we would need to make sure, that all parent directory permissions are safe (preventing a directory from being replaced by a less privileged malicious plugin).
This is probably doable in a unix environment, but maybe would be very hard / a nightmare on a platform like Windows (but I do know the relevant details).

Should the removal afterwards be based on the file handle that we opened before for this directory? Or should we use the path instead. The file handle would probably be safer - but harder/impossible to implement on other platforms.

I have the feeling, that the safe cleanup of such a directory would end up being a source of security issues, while providing only very little convenience.
But maybe I underestimate the benefits and overestimate the implementation problems.

**Comment by steveschnepp** [comment]
Also, I don't feel very comfortable about providing many options to secure it very tightly.

As it is rather complicated and therefore time consuming. Also I don't like the false sense of security it conveys.

I mean, one should only use plugins one trusts. That is why we advocate shell & scripts. 

Compiled ones should be exceptional, and merely done by us.

**Comment by sumpfralle** [comment]
> I mean, one should only use plugins one trusts.

Yes, this would be good. But on the other hand, users can expect, that a plugin running with limited privileges (e.g. as user "munin") cannot gain more privileges by interfering with the files or directories that are accessed by a root-user plugin (by accident or on purpose).
(just theoretical thinking - without a specific example)

Anyway: I do not feel entitled to stop you :)
I would just be a bit scared to implement the directory cleanup myself.

**Comment by sumpfralle** [comment]
Thinking further about it: if we are currently comtemplating about switching our default `munin-node` implementation from the one shipped with `munin` to `munin-node-c`, then I would recommend to keep the interface that is required by a `munin-node` implementation to be as minimal and simple, as possible. Otherwise we reduce the usefulness of alternative existing implementations (e.g. `muninlite` - which is widely used on openwrt).
